quickwit-storage/src/split.rs

```diff
@@ -0,0 +1,379 @@
+    range: Range<usize>,
+) -> io::Result<(usize, usize)> {
+    let get_start = blocks_ranges
+        .iter()
+        .position(|block_range| block_range.end > range.start)
+        .ok_or_else(|| io::Error::new(ErrorKind::InvalidInput, "range is out of Range"))?;
+    let get_end = blocks_ranges
+        .iter()
+        .position(|block_range| block_range.end >= range.end)
+        .ok_or_else(|| io::Error::new(ErrorKind::InvalidInput, "range is out of Range"))?;
+    Ok((get_start, get_end))
+}
+
+/// SplitOffsetComputer is used to bundle together multiple files into a single split file.
+#[derive(Default, Debug)]
+pub struct SplitOffsetComputer {
+    metadata: BundleStorageFileOffsets,
+    current_offset: usize,
+}
+
+impl SplitOffsetComputer {
+    /// Creates a new SplitOffsetComputer, to which files and the hotcache can be added.
+    pub fn new() -> Self {
+        SplitOffsetComputer {
+            current_offset: 0,
+            metadata: Default::default(),
+        }
+    }
+
+    /// Adds the file to the bundle file.
+    ///
+    /// The hotcache needs to be the last file that is added, in order to be able to read
+    /// the hotcache and the metadata in one continous read.
+    pub fn add_file(&mut self, path: &Path) -> io::Result<()> {
+        let file = std::fs::metadata(path)?;
+        let file_range = self.current_offset as u64..self.current_offset as u64 + file.len() as u64;
+        self.current_offset += file.len() as usize;
+        self.metadata.files.insert(path.to_owned(), file_range);
+        Ok(())
+    }
+
+    /// Writes the bundle file offsets metadata at the end of the bundle file,
+    /// and returns the byte-range of this metadata information.
+    pub fn finalize(self, hotcache: &[u8]) -> io::Result<SplitStreamer> {
+        // Build the footer.
+        let mut footer_bytes = vec![];
+        // Fix paths to be relative
+        let metadata_with_fixed_paths = self
+            .metadata
+            .files
+            .iter()
+            .map(|(path, range)| {
+                let file_name = PathBuf::from(path.file_name().ok_or_else(|| {
+                    io::Error::new(
+                        ErrorKind::InvalidInput,
+                        format!("could not extract file_name from path {:?}", path),
+                    )
+                })?);
+                Ok((file_name, range.start..range.end))
+            })
+            .collect::<Result<HashMap<_, _>, io::Error>>()?;
+
+        let metadata_json = serde_json::to_string(&BundleStorageFileOffsets {
+            files: metadata_with_fixed_paths,
+        })?;
+
+        footer_bytes.write_all(metadata_json.as_bytes())?;
+        let metadata_json_len = metadata_json.len() as u64;
+        footer_bytes.write_all(&metadata_json_len.to_le_bytes())?;
+        footer_bytes.write_all(hotcache)?;
+        footer_bytes.write_all(&hotcache.len().to_le_bytes())?;
+
+        let mut files_and_offsets = self
```

Comment:
`block_offsets` and `files_and_offsets` seem redundant, maybe if we manage to simplify the `range_byte_stream` we will manage to simplify this.

Reply:
it's redundant, we could use files_and_offsets, but like this it allows for a minimal api when searching for the block